src/operators/custom_version.rs

```diff
@@ -0,0 +1,114 @@
+use std::cmp::Ordering;
+use std::collections::VecDeque;
+use unicode_normalization::UnicodeNormalization;
+
+fn split_version(version: &str) -> VecDeque<String> {
+    let mut components = VecDeque::new();
+    let mut part = String::new();
+    for ch in version.chars() {
+        if ch == '.' || ch == '-' {
+            components.push_back(part.drain(..).collect());
+            part = String::new();
+        } else {
+            part.push(ch);
+        }
+    }
+    components.push_back(part.drain(..).collect());
+    components
+}
+
+pub fn compare_version(str1: &str, str2: &str, numeric: bool) -> Ordering {
+    let norm_str1 = str1.nfc().collect::<String>();
+    let norm_str2 = str2.nfc().collect::<String>();
+    let mut iter1 = split_version(&norm_str1);
+    let mut iter2 = split_version(&norm_str2);
+
+    loop {
```

Comment:
Can we instead iterate over both queues? Infinity loops with a break somewhere aren't advised